[IMP] web, *: remove explicit references to `session_id`

mge-odoo opened the pull request:
Before this rev. the `session_id` was provided by the server to the webclient (in
session_info, mostly) and was stored and accessible. This made XSS injection more
dangerous than they should be as it was very easy to steal the `session_id`.

As the browser automatically set the `session_id` on every request to the server,
the webclient shouldn't need any explicit reference.

Since rev. odoo/odoo@f4d541e the `session_id` cookie uses the `httponly` flag so
it cannot be accessed through client side script.

Task 1905359

antonylesuisse commented:
I fully agree lets get rid of this historical artifact.

For the context, I included jsonp support to allow embedding odoo action on a external website (like embbed a video) a feature that was actually never used. Then it was still used for the apps action in v7 and maybe v8.

BUT we need to check to make sure that livechat on an external website, a feature still supported, still work. Or maybe we could get rid of that feature too.

mge-odoo commented:
@antonylesuisse @odony the livechat on an external website now works with CORS instead of JSONP. Note that it implies to allow cross-origin on some routes (`cors="*"`), including the `/poll` and `/mail/post`.
I have modified some rpcs to avoid allowing cross-origin on `/web/proxy/load` (security precaution) and I have removed this route (the only use case was the livechat one, see https://github.com/odoo/odoo/commit/7f8fe1076660ab949a050c04443abd160555e08a and https://github.com/odoo/odoo/commit/f4bb9bc9762c2fa99d3f72126eb0efc9ace1afa4).

mge-odoo commented:
@odony just a kind reminder for you to check the `/mail/post` route for security issues before I merge and the use of `/web/proxy/load`.

mge-odoo commented:
@odony no pressure but if you want this task to make it through this saas version, it's probably better to take a look before the end of the week :)

odony commented:
Mmm, github is broken and I can't post inline review comments.

So the comment below is for [this line](https://github.com/odoo/odoo/pull/30098/files#diff-a968dc717b55409ac0fa73e06c3c070cR44)

----
This inner `werkzeug.test.Client` looks oddish and has [its origin almost lost in history](75409e349a0ca4488e477b9621991145e6890d5c), from old times where:
- the web client was a separate, single-threaded program
- the proxy was supposed to allow arbitrary requests

7 years have passed and we're repurposing the code to only load a few XML files ;-)
Could we simplify it and directly load the file contents? Theoretically all it takes is `tools.file_open('module/path/to/file.xml', 'rb').read()` (!no leading slash in the path).
The XML files are small and the list of files is hardcoded (good!) so there is little risk of memory bloat or abuse here.

ged-odoo commented:
robodoo r+

robodoo commented:
Linked pull request(s) odoo/enterprise#3405 not ready. Linked PRs are not staged until all of them are ready.

robodoo commented:
Because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

mge-odoo commented:
robodoo rebase-merge

robodoo commented:
I'm sorry, @mge-odoo. You can't method=rebase-merge.

aab-odoo commented:
robodoo rebase-merge

robodoo commented:
Merge method set to rebase and merge, using the PR as merge commit message

robodoo commented:
Merged, thanks!

mge-odoo commented:
@odony note that while grepping `jsonpRpc`, there were still occurrences in internal repository that would probably require to be adapted.

odony commented:
> @odony note that while grepping `jsonpRpc`, there were still occurrences in internal repository that would probably require to be adapted.

Good point, task 1940573 will take care of it, thanks :-)
